connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/v2/V2JDBCTest.scala

```diff
@@ -986,4 +986,23 @@ private[v2] trait V2JDBCTest extends SharedSparkSession with DockerIntegrationFu
   test("scan with filter push-down with date time functions") {
     testDatetime(s"$catalogAndNamespace.${caseConvert("datetime")}")
   }
+
+  test("SPARK-50792 Format binary data as a binary literal in JDBC.") {
+    val tableName = s"$catalogName.test_binary_literal"
+    withTable(tableName) {
+      // Unfornately, Oracle can only compare two BLOBs with a special function dbms_lob.compare
+      // The V2ExpressionSQLBuilder can not support rewriting the '=' operator.
+      // We don't test binary data on Oracle and do not support binary data on Oracle.
+      if (!this.isInstanceOf[OracleIntegrationSuite]) {
```

Comment:
The test case skips the Oracle, but the users still possible use this case.

Reply:
Yes, that requires a big change in how V2ExpressionSQLBuilder works. The binary literal was not working before this PR anyway. We can make it work at least on other databases this time. And propose another design changes to V2ExpressionSQLBuilder to make it flexible enable to adopt cases like this.

Reply:
That's not a good idea if the bug still exists when using oracle dialect. Is there a way to fix the bug?

Reply:
> Yes, that requires a significant change in how V2ExpressionSQLBuilder works. The binary literal was not working before this PR anyway. We can make it work at least on other databases this time. And propose another design changes to V2ExpressionSQLBuilder to make it flexible enable to support cases like this.

Why we need change V2ExpressionSQLBuilder? Please describe the detail.

Reply:
No, we don't need to change V2ExpressionSQLBuilder. It's all in OracleDialect now.

Reply:
BTW: I just changed test to verify the returned result in addition. It's going to run for a while.

Reply:
All tests passed, but downloading report failed. We can rerun the whole test again to clear all the checks, but it takes quite some time to finish. 
https://github.com/sunxiaoguang/spark/actions/runs/12744731853